docs/content/docs/dev/table/sql/queries/hints.md

```diff
@@ -84,4 +84,244 @@ insert into kafka_table1 /*+ OPTIONS('sink.partitioner'='round-robin') */ select
+SELECT /*+ BROADCAST(t1) */ * FROM view1 JOIN t1 ON view1.id = t1.id;
+
+
+-- Conflict in One Same Join Hints Strategy Case
+
+-- For this conflict in one same Join Hints strategy case, the first hint will be chosen, t2 will be the broadcast table.
+SELECT /*+ BROADCAST(t2), BROADCAST(t1) */ * FROM t1 JOIN t2 ON t1.id = t2.id;
+
+-- For this conflict in one same Join Hints strategy case, t1 will be the broadcast table.
+SELECT /*+ BROADCAST(t1, t1) */ * FROM t1 JOIN t2 ON t1.id = t2.id;
+
+-- For this conflict in one same Join Hints strategy case, t2 will be the broadcast table.
+SELECT /*+ BROADCAST(t2, t1), BROADCAST(t1, t2) */ * FROM t1 JOIN t2 ON t1.id = t2.id;
+
+-- For this conflict in one same Join Hints strategy case, it equals BROADCAST(t1, t2) + BROADCAST(t3),
+-- when join between t1 and t2, t1 will be the broadcast table,
+-- when join between the result after t1 join t2 and t3, t3 will be the broadcast table.
+SELECT /*+ BROADCAST(t1, t2, t3) */ * FROM t1 JOIN t2 ON t1.id = t2.id JOIN t3 ON t1.id = t3.id;
+
+
+-- Conflict in Different Join Hints Strategies Case
+
+-- For this conflict in different Join Hints strategies case, an exception will be thrown when parse this query statement.
+SELECT /*+ BROADCAST(t1) */ /*+ SHUFFLE_HASH(t1) */ * FROM t1 JOIN t2 ON t1.id = t2.id;
+
+-- For this conflict in different Join Hints strategies case, the first Join Hints strategy (BROADCAST Join Hints) will be chosen.
+SELECT /*+ BROADCAST(t1) SHUFFLE_HASH(t1) */ * FROM t1 JOIN t2 ON t1.id = t2.id;
+
+-- For this conflict in different Join Hints strategies case, although BROADCAST Join Hints is first, but it doesn't support full outer join,
+-- so choose the following SHUFFLE_HASH Join Hints.
+SELECT /*+ BROADCAST(t1) SHUFFLE_HASH(t1) */ * FROM t1 FULL OUTER JOIN t2 ON t1.id = t2.id;
+
+-- For this conflict in different Join Hints strategies case, although there are two Join Hints defined, but all of them are neither support non-equivalent join,
+-- so default strategy will be chosen.
+SELECT /*+ BROADCAST(t1) SHUFFLE_HASH(t1) */ * FROM t1 FULL OUTER JOIN t2 ON t1.id > t2.id;
+```
+
+## What are query blocks ?
+
+A `query block` is a basic unit of SQL. For example, any inline view or subQuery of a SQL statement are considered separate
+`query block` to the outer query.
+
+### Examples
+
+An SQL statement can consist of several subQuery. The subQuery can be a `SELECT`, `INSERT` or `DELETE`. A subQuery can contain
+other subQueries in the `FROM` clause, the `WHERE` clause, or a subSelect of a `UNION` or `UNION ALL`.
+
+For these different subQuery type, they can be composed of several `query block`, For example:
+
+The simple query in below picture is a subQuery, contains by two `query blocks` - one for the `SELECT` in the `FROM` clause and one
+for the outer `SELECT`.
+
+{{< img src="/fig/hint/hint_query_block_from.png" alt="hint from query block" >}}
```

Comment:
there is no much difference for the first two pictures, we can merge them into one